src/prefect/engine.py

```diff
@@ -2056,6 +2065,32 @@ def should_log_prints(flow_or_task: Union[Flow, Task]) -> bool:
     return flow_or_task.log_prints
 
 
+def _resolve_custom_flow_run_name(flow: Flow, parameters: Dict[str, Any]) -> str:
+    if callable(flow.flow_run_name):
+        flow_run_name = flow.flow_run_name()
+        if not isinstance(flow_run_name, str):
+            raise RuntimeError("callable 'flow_run_name' did not return a string")
+    elif isinstance(flow.flow_run_name, str):
+        flow_run_name = flow.flow_run_name.format(**parameters)
+    else:
+        raise RuntimeError("'flow_run_name' is neither a string nor a callable")
+
+    return flow_run_name
+
+
+def _resolve_custom_task_run_name(task: Task, parameters: Dict[str, Any]) -> str:
+    if callable(task.task_run_name):
+        task_run_name = task.task_run_name()
+        if not isinstance(task_run_name, str):
+            raise RuntimeError("callable 'task_run_name' did not return a string")
```

Comment:
We probably want something more verbose for debugging

```suggestion
            raise RuntimeError(f"Callable {task.task_run_name} for 'task_run_name' returned type {type(task_run_name).__name__} but a string is required.")
```

Similar for the other error messages here

Reply:
done